# **This colab file is only for data Augmentation**

---



#Augmentation of present 2459 sample size of dataset

In [4]:
#!pip install nlpaug
#!pip install pandas
#!pip install torch torchvision torchaudio
#!pip install transformers



In [5]:
import nltk

# Download required resources
nltk.download('punkt')
nltk.download('averaged_perceptron_tagger_eng')
nltk.download('wordnet')
nltk.download('omw-1.4')


[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package averaged_perceptron_tagger_eng to
[nltk_data]     /root/nltk_data...
[nltk_data]   Package averaged_perceptron_tagger_eng is already up-to-
[nltk_data]       date!
[nltk_data] Downloading package wordnet to /root/nltk_data...
[nltk_data]   Package wordnet is already up-to-date!
[nltk_data] Downloading package omw-1.4 to /root/nltk_data...
[nltk_data]   Package omw-1.4 is already up-to-date!


True

In [6]:
#checking original dataset size

import pandas as pd
df = pd.read_csv("/content/drive/MyDrive/SEM_3_Project/PhishingEmailData_1.csv", encoding='latin1')
print("Original rows:", len(df))
print("Shape:", df.shape)


Original rows: 189
Shape: (189, 13)


In [7]:
!pip install nlpaug

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 410.5/410.5 kB 10.3 MB/s eta 0:00:00


In [9]:
import nlpaug.augmenter.word as naw
import pandas as pd

# Load dataset
df = pd.read_csv("/content/drive/MyDrive/SEM_3_Project/PhishingEmailData_1.csv", encoding='latin1')

# Create synonym augmenter
aug = naw.SynonymAug(aug_src='wordnet')

# Function to augment specific text columns
def augment_text_columns(row):
    new_row = row.copy()
    for col in ['Email_Subject', 'Email_Content', 'Closing_Remarks']:
        if pd.notna(row[col]):  # Check if value is not NaN
            new_row[col] = aug.augment(str(row[col]))
    return new_row

# Apply augmentation for all rows
augmented_rows = df.apply(augment_text_columns, axis=1)

# Combine with original
final_df = pd.concat([df, augmented_rows])

# Save new dataset
final_df.to_csv("emails_augmented.csv", index=False)


###check the augmented new dataset below

In [10]:
import pandas as pd
df_new = pd.read_csv("emails_augmented.csv")
print(df_new.shape)
print(df_new.head(10))


(378, 13)
                                       Email_Subject  \
0                                     URGENT REQUEST   
1                                     Quick question   
2   ******Part time home work assistant needed******   
3                                   Ê vendor payment   
4                                     Quick question   
5  ÊXXX has shared a document on Google Docs with...   
6                       Message from human resources   
7                                    Library Account   
8                                  Your Dropbox File   
9                                                NaN   

                                       Email_Content Sending_Date  \
0  Are you available ?\nNo calls text only 951307...       1/9/20   
1  I'm in a meeting and need help getting some Am...       1/9/20   
2  Hello RECIPIENT\n\nI am urgently seeking for a...     10/19/19   
3  Are you around? I need to pay a vendor with th...  12/27/18\n    
4  I'm in a meeting and need

In [11]:
#checking size of dataset
# Show number of rows
print("Number of rows:", len(df))

# Or using shape (rows, columns)
print("Shape of dataset:", df.shape)

Number of rows: 189
Shape of dataset: (189, 13)


In [12]:
augmented_rows = df.apply(augment_text_columns, axis=1)


failed in augmentation as our base size was small 189 and yet after its same now we are trying again with another method

In [13]:
import nlpaug.augmenter.word as naw
import pandas as pd

# Load dataset
df = pd.read_csv("/content/drive/MyDrive/SEM_3_Project/PhishingEmailData_1.csv", encoding='latin1')
print("Original dataset size:", df.shape)

# Synonym augmenter
aug = naw.SynonymAug(aug_src='wordnet')

# Create augmented dataset
augmented_rows = []

for i, row in df.iterrows():
    new_row = row.copy()
    for col in ['Email_Subject', 'Email_Content', 'Closing_Remarks']:
        if pd.notna(row[col]):
            new_row[col] = aug.augment(str(row[col]))
    augmented_rows.append(new_row)

# Convert augmented rows to dataframe
df_aug = pd.DataFrame(augmented_rows)

# Combine original + augmented
final_df = pd.concat([df, df_aug], ignore_index=True)

print("Final dataset size:", final_df.shape)

# Save
final_df.to_csv("emails_augmented.csv", index=False)


Original dataset size: (189, 13)
Final dataset size: (378, 13)


fine the data got double but our target is to have dataset more than 4000 size for that we do

In [15]:
import nlpaug.augmenter.word as naw
import pandas as pd

# Load your dataset
df = pd.read_csv("/content/drive/MyDrive/SEM_3_Project/PhishingEmailData_1.csv", encoding='latin1')

# Initialize augmenter (synonym replacement)
aug = naw.SynonymAug(aug_src='wordnet')

# Desired dataset size
target_size = 5000

# Copy original
augmented_df = df.copy()

# Keep augmenting until we reach the target
while len(augmented_df) < target_size:
    new_data = df.copy()

    # Apply augmentation on text column (replace 'email_text' with your actual column)
    new_data['Email_Subject'] = new_data['Email_Subject'].apply(lambda x: aug.augment(str(x)))
    new_data['Email_Content'] = new_data['Email_Content'].apply(lambda x: aug.augment(str(x)))
    new_data['Closing_Remarks'] = new_data['Closing_Remarks'].apply(lambda x: aug.augment(str(x)))

    augmented_df = pd.concat([augmented_df, new_data], ignore_index=True)

# Trim extra rows if it exceeds 5000
augmented_df = augmented_df.head(target_size)

print("Original dataset size:", df.shape)
print("Final dataset size:", augmented_df.shape)

# Save final dataset
augmented_df.to_csv("augmented_dataset.csv", index=False)


Original dataset size: (189, 13)
Final dataset size: (5000, 13)


Based on above result ourprevious data frm 189 rows and 13 columns we augmented same dataset to** 5000 rows of data and 13 columns**


---


check the path of the file down 👇

In [16]:
import os

file_path = "augmented_dataset.csv"
print(f"The augmented dataset is saved at: {os.path.abspath(file_path)}")

The augmented dataset is saved at: /content/augmented_dataset.csv


download the file into PC 👇

In [17]:
from google.colab import files

files.download('augmented_dataset.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

#now since our data is around 5000 we can perform

#END HERE


---

